src/service/viewer-impl.js

```diff
@@ -872,8 +872,16 @@ export class Viewer {
       return Promise.resolve('');
     }
     return this.sendMessageUnreliable_('fragment', undefined, true).then(
-      hash => hash || ''
-    );
+        hash => {
+          if (!hash) {
+            return '';
+          }
+          /* Strip leading '#' */
+          if (hash.indexOf('#') == 0) {
```

Comment:
Meaning they don't have to call `#substr`, or even remember that a manipulation is needed. That makes it simpler for htem.

If we go with leading `#`, then we should assert the string starts with a `#`. If we go without leading `#`, then assert it doesn't.